sql/core/src/test/scala/org/apache/spark/sql/ShowCreateTableSuite.scala

```diff
@@ -155,16 +155,17 @@ abstract class ShowCreateTableSuite extends QueryTest with SQLTestUtils {
       val ex = intercept[AnalysisException] {
         sql(s"SHOW CREATE TABLE $viewName")
       }
-      assert(ex.getMessage.contains("SHOW CREATE TABLE is not supported on a temporary view"))
+      assert(ex.getMessage.contains(s"$viewName is a temp view not table or permanent view"))
```

Comment:
Not related to this PR: One idea to improve the error message is to put the command name in `UnresolvedTableOrView` or other similar plans when the parser creates the command logical plan.

Reply:
A good idea! I will follow up on this.